### PR TITLE
[CWS-531] - Introducing the heartbeat event

### PR DESCRIPTION
**Pull request by Malo10LeGoff**
### What does this PR do?

Generate a heartbeat event per policy every minute that is composed of : 
- The policy name
- The policy source : RC or local
- The policy version

### Motivation

Improving the visibility of the customers of the state of their fleet

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/rules/policy_monitor.go [hunk 1]
```diff
@@ -244,3 +251,27 @@ func NewRuleSetLoadedEvent(ruleSets map[string]*rules.RuleSet, err *multierror.E
...
+func NewHeartbeatEvents(policies map[string]Policy) (*rules.Rule, []*events.CustomEvent) {
+
+	var evts []*events.CustomEvent
+
+	for _, policy := range policies {
+		var policyState = PolicyState{
+			Name:    policy.Name,
+			Version: policy.Version,
+			Source:  policy.Source,
+			Rules:   nil, // The rules that have been loaded at startup are not reported in the heartbeat event
```
Comment (Malo10LeGoff) [inline pkg/security/rules/policy_monitor.go:265]: Very open to discussion but I'm thinking it's not super useful to include all the rules in every heartbeat event knowing it's already sent in ruleset_loaded

File: pkg/security/rules/engine.go [hunk 2]
```diff
@@ -183,6 +183,24 @@ func (e *RuleEngine) Start(ctx context.Context, reloadChan <-chan struct{}, wg *
 			}
 		}
 	}()
+
+	// Sending an heartbeat event every minute
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		heartbeatTicker := time.NewTicker(1 * time.Minute)
```
Comment (Malo10LeGoff) [inline pkg/security/rules/engine.go:192]: The frequency at which the heartbeat event is generated 1 minutes, let me know if you disagree, that might be too much

File: pkg/security/rules/engine.go [hunk 3]
```diff
@@ -183,6 +183,24 @@ func (e *RuleEngine) Start(ctx context.Context, reloadChan <-chan struct{}, wg *
 			}
 		}
 	}()
+
+	// Sending an heartbeat event every minute
+	wg.Add(1)
```
Comment (safchain) [inline pkg/security/rules/engine.go:188]: could we move the heartbeat thing in the policy monitor ?

Comment (Malo10LeGoff) [reply pkg/security/rules/engine.go:188]: Done in https://github.com/DataDog/datadog-agent/pull/19775/commits/46019af51cb7346badfc4354ae90e5c4e6a756a3

**Review by safchain — APPROVED** [review]

**Pull request merged**